Tests/PublicAPISpec.swift

```diff
@@ -119,15 +118,58 @@ class PublicAPISpec: QuickSpec {
                 RInAppMessaging.deinitializeModule()
-                reinitializeSDK()
-                messageMixerService.delay = 3.0
+                reinitializeSDK(onDependencyResolved: {
+                    messageMixerService.delay = 3.0
+                })
                 RInAppMessaging.logEvent(LoginSuccessfulEvent())
                 expect(eventMatcher.loggedEvents).toAfterTimeout(beEmpty(), timeout: 1)
                 expect(eventMatcher.loggedEvents).toEventually(haveCount(1),
                                                                timeout: .seconds(Int(messageMixerService.delay + 1)),
                                                                pollInterval: .milliseconds(500))
             }
 
+            // This test checks if events logged after getConfig() failure (bufferedEvents)
+            // are re-logged properly when retried getConfig request succeeds.
+            // If buffered events can fill the set of campaign triggers, for example 3 times,
+            // then that campaign should be displayed 3 times.
+            // This test also checks if campaigns are queued properly (respecting impressionsLeft value)
+            // when multiple events are logged in very short time.
+            it("will display campaigns after first ping call for matching events that were logged after getConfig request failure") {
+                RInAppMessaging.deinitializeModule()
+                reinitializeSDK(onDependencyResolved: {
+                    configurationManager.simulateRetryDelay = 1.0
+                    messageMixerService.delay = 1.0
+                    messageMixerService.mockedResponse = TestHelpers.MockResponse.withGeneratedCampaigns(
+                        count: 1, test: false, delay: 100, addContexts: false,
+                        triggers: [Trigger(type: .event,
+                                           eventType: .loginSuccessful,
+                                           eventName: "e1",
+                                           attributes: [])])
+                    // maxImpressions = 2
+                })
+                RInAppMessaging.logEvent(LoginSuccessfulEvent())
+                RInAppMessaging.logEvent(LoginSuccessfulEvent())
+                RInAppMessaging.logEvent(LoginSuccessfulEvent())
+
+                expect(UIApplication.shared.keyWindow?.subviews).toEventually(containElementSatisfying({
+                    if let view = $0 as? BaseView {
+                        view.dismiss()
+                        return true
+                    }
+                    return false
+                }), timeout: .seconds(3))
+                expect(UIApplication.shared.keyWindow?.subviews).toEventually(containElementSatisfying({
+                    if let view = $0 as? BaseView {
+                        view.dismiss()
+                        return true
+                    }
+                    return false
+                }))
```

Comment:
So there are 2 expect(contain) because maxImpressions is 2?

Reply:
yes, the last expect checks if it doesn't exceed max impression (because there are 3 LoginEvents)

Reply:
I understand it now! 🥳